File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        HashMap<Integer, Integer> counts = new HashMap<>();
        for (int num: nums) {
            if (!counts.containsKey(num)) {
                counts.put(num, 0);
            }
            counts.put(num, counts.get(num) + 1);
            if (counts.get(num) > 1) {
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # nums=[1,2,2,3,3,3]
        # k=2
        # frequency: [[], [], ..., []] where i (count) -> values
        # max count is the length of nums i.e. 6, range(7), which will be 0, 1, ..., 6

        counts = {}

        for num in nums:
            counts[num] = counts.get(num, 0) + 1 # num -> count {1: 1, 2: 2, 3: 3}
        
        size = len(nums)
        frequency = [[] for _ in range(size + 1)]

        for num, count in counts.items():
            frequency[count].append(num)  # frequency: [[], [1], [2], [3], [], [], []]
        
        print(frequency)

        result = []
        for i in range(len(frequency) - 1, 0, -1):
            for num in frequency[i]:
                result.append(num)
                if len(result) == k:
                    return result
                




        

        